### PR TITLE
[EventGrid] Readme, JavaDoc, Samples For 2.0.0

### PR DESCRIPTION
**Pull request by YijunXieMS**
1. Update JavaDoc of `EventGridEventPublisherClient`, `EventGridEventPublisherAsyncClient`
2. Update Samples and README after CloudEvent is moved to moved to azure-core
3. Bring back `EventGridPublisherClientBuilder.serviceVersion()`

### CODE REVIEW

File: sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridPublisherAsyncClient.java [hunk 1]
```diff
@@ -44,10 +44,23 @@
 
 /**
  * A service client that publishes events to an EventGrid topic or domain. Use {@link EventGridPublisherClientBuilder}
- * to create an instance of this client. This uses Project Reactor (https://projectreactor.io/) to handle asynchronous
- * programming.
+ * to create an instance of this client. It uses <a href="https://projectreactor.io/">Project Reactor</a> to
+ * handle asynchronous programming.
```
Comment (srnagar) [inline sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridPublisherAsyncClient.java:48]: I don't think mentioning project reactor is necessary here.

File: sdk/eventgrid/azure-messaging-eventgrid/src/samples/java/com/azure/messaging/eventgrid/samples/EventGridEventJavaDocCodeSnippet.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        // Create a client to send events of EventGridEvent schema
+        EventGridPublisherClient<EventGridEvent> eventGridEventPublisherClient = new EventGridPublisherClientBuilder()
+            .endpoint(System.getenv("AZURE_EVENTGRID_EVENT_ENDPOINT"))  // make sure it accepts EventGridEvent
+            .credential(new AzureKeyCredential(System.getenv("AZURE_EVENTGRID_EVENT_KEY")))
+            .buildEventGridEventPublisherClient();
+        // END: com.azure.messaging.eventgrid.EventGridPublisherClient#CreateEventGridEventClient
+
+        // BEGIN: com.azure.messaging.eventgrid.EventGridPublisherClient#SendEventGridEvent
+        // Create an EventGridEvent
+        User user2 = new User("John", "James");
```
Comment (srnagar) [inline sdk/eventgrid/azure-messaging-eventgrid/src/samples/java/com/azure/messaging/eventgrid/samples/EventGridEventJavaDocCodeSnippet.java:162]: `user2` in javadoc of a separate method may not be ideal. Split this javadoc snippet into a new method and name this as `user` instead.

File: sdk/eventgrid/azure-messaging-eventgrid/src/samples/java/com/azure/messaging/eventgrid/samples/EventGridEventJavaDocCodeSnippet.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+        // Send a list of EventGridEvents to the EventGrid service altogether.
+        // This has better performance than sending one by one.
+        eventGridEventPublisherClient.sendEvents(new ArrayList<EventGridEvent>() {
+            {
+                add(eventGridEvent);
+                // add more EventGridEvents objects
+            }
+        });
+        // END: com.azure.messaging.eventgrid.EventGridPublisherClient#SendEventGridEvent
```
Comment (srnagar) [inline sdk/eventgrid/azure-messaging-eventgrid/src/samples/java/com/azure/messaging/eventgrid/samples/EventGridEventJavaDocCodeSnippet.java:177]: Do we have snippets for publishing custom events too?

**Review by srnagar — APPROVED** [review]
LGTM!

File: sdk/eventgrid/azure-messaging-eventgrid/src/samples/java/com/azure/messaging/eventgrid/samples/EventGridEventJavaDocCodeSnippet.java (hunk 3 above)
Comment (YijunXieMS) [reply sdk/eventgrid/azure-messaging-eventgrid/src/samples/java/com/azure/messaging/eventgrid/samples/EventGridEventJavaDocCodeSnippet.java:219]: Added snippets for publishing custom events

**Pull request merged**